runtime/src/iree/hal/drivers/hip/hip_device.c

```diff
@@ -922,6 +966,39 @@ iree_hal_hip_device_query_semaphore_compatibility(
   return IREE_HAL_SEMAPHORE_COMPATIBILITY_HOST_ONLY;
 }
 
+static iree_status_t iree_hal_hip_device_pepare_async_alloc(
+    iree_hal_hip_device_t* base_device, iree_hal_buffer_params_t params,
+    iree_device_size_t allocation_size,
+    iree_hal_buffer_t** IREE_RESTRICT out_buffer) {
+  IREE_TRACE_ZONE_BEGIN(z0);
+  IREE_TRACE_ZONE_APPEND_VALUE_I64(z0, (int64_t)allocation_size);
+
+  iree_hal_buffer_params_canonicalize(&params);
+
+  // NOTE: we don't provide a device allocator because we didn't allocate from
```

Comment:
without a release callback or an allocator the buffer can never make it back to the allocator? (no release callback is provided below).